File: backend/app/services/gold_economy.py
"""Баланс золота: привязка к «Часам Фокуса» и сложности квеста."""

from app.models.enums import QuestDifficulty

# 1 час отдыха в магазине = 20 золота; 2:1 труд/отдых.
GOLD_PER_FOCUS_HOUR = 10

GOLD_BY_DIFFICULTY: dict[QuestDifficulty, int] = {
    QuestDifficulty.TRIVIAL: 2,      # ~5 мин
    QuestDifficulty.EASY: 5,           # 15–30 мин
    QuestDifficulty.NORMAL: 10,      # 1 час фокуса (Medium)
    QuestDifficulty.HARD: 25,        # 2–3 часа
    QuestDifficulty.LEGENDARY: 60,     # целый день (Epic)
}


def gold_for_difficulty(difficulty: QuestDifficulty) -> int:
    return GOLD_BY_DIFFICULTY.get(difficulty, GOLD_BY_DIFFICULTY[QuestDifficulty.NORMAL])


def gold_for_rest_hours(hours: float) -> int:
    """Стоимость награды-отдыха: 2 часа работы = 1 час отдыха."""
    return max(1, round(hours * GOLD_PER_FOCUS_HOUR * 2))

File: frontend/src/lib/gold-economy.ts
import type { QuestDifficulty } from "@/lib/types";

/** 1 час отдыха в магазине = 20 золота (2:1 труд/отдых). */
export const GOLD_PER_FOCUS_HOUR = 10;

export const GOLD_BY_DIFFICULTY: Record<QuestDifficulty, number> = {
  trivial: 2,
  easy: 5,
  normal: 10,
  hard: 25,
  legendary: 60,
};

export function goldForDifficulty(difficulty: QuestDifficulty): number {
  return GOLD_BY_DIFFICULTY[difficulty] ?? GOLD_BY_DIFFICULTY.normal;
}

export function goldForRestHours(hours: number): number {
  return Math.max(1, Math.round(hours * GOLD_PER_FOCUS_HOUR * 2));
}

export const DIFFICULTY_GOLD_LABELS: Record<QuestDifficulty, string> = {
  trivial: "2 🪙 (5 мин)",
  easy: "5 🪙 (15–30 мин)",
  normal: "10 🪙 (1 ч фокуса)",
  hard: "25 🪙 (2–3 ч)",
  legendary: "60 🪙 (весь день)",
};
